### PR TITLE
Replace parameter unicodeSetFilter with unicode_set_filter 

### LINKED ISSUE #22823: IcuFoldingTokenFilter still accepts camelCase parameter
https://github.com/elastic/elasticsearch/blob/master/plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuFoldingTokenFilterFactory.java#L48

https://www.elastic.co/guide/en/elasticsearch/plugins/current/analysis-icu-folding.html

### PR DESCRIPTION
**Pull request by liketic**
Depreate parameter unicodeSetFilter and replace it with unicode_set_filter for IcuFoldingTokenFilter. 

Close #22823

### CODE REVIEW

**Comment by javanna** [comment]
@elastic/es-search-aggs thoughts on this?

**Comment by colings86** [comment]
@romseygeek would you be able to pick up reviewing this PR?

**Comment by romseygeek** [comment]
I think this looks good, @liketic would you be able to update this to the latest master and push again?

**Comment by liketic** [comment]
Thanks @romseygeek , I updated the branch, please review again.

**Comment by romseygeek** [comment]
@elasticmachine test this please

**Comment by romseygeek** [comment]
Looks good, thanks @liketic.  One more thing, could you update `docs/plugins/analysis-icu.asciidoc` to replace all instances of `unicodeSetFilter` with `unicode_set_filter`?

**Comment by romseygeek** [comment]
Thanks @liketic !

File: plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuNormalizerTokenFilterFactory.java [hunk 1]
```diff
@@ -35,14 +38,15 @@
  * <p>The {@code unicodeSetFilter} attribute can be used to provide the UniCodeSet for filtering.</p>
  */
 public class IcuNormalizerTokenFilterFactory extends AbstractTokenFilterFactory implements MultiTermAwareComponent {
-
+    private final static DeprecationLogger deprecationLogger =
```
Comment (atorok) [inline plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuNormalizerTokenFilterFactory.java]: @romseygeek this line is causing checkstyle violations
```
> Task :plugins:analysis-icu:checkstyleMain FAILED
[ant:checkstyle] [ERROR] /home/alpar/work/elastic/elasticsearch/plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuNormalizerTokenFilterFactory.java:41:19: 'static' modifier out of order with the JLS suggestions. [ModifierOrder]
```
I'm pushing a fix to master now. FYI

Comment (atorok) [reply plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuNormalizerTokenFilterFactory.java]: Master was merged in a few days ago and the checkstyle rules last changed 4 weeks ago, yet the PR build passed.

Comment (romseygeek) [reply plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuNormalizerTokenFilterFactory.java]: I opened #35207 to fix these

**Pull request merged**